modules/cms/classes/Controller.php

```diff
@@ -1436,31 +1436,52 @@ public function param($name, $default = null)
      * @param array  $properties  Component properties
      * @param bool   $addToLayout Add to layout, instead of page
      * @return ComponentBase Component object
+     * @throws CmsException
+     * @throws SystemException
      */
     public function addComponent($name, $alias, $properties, $addToLayout = false)
     {
         $manager = ComponentManager::instance();
 
-        if ($addToLayout) {
-            if (!$componentObj = $manager->makeComponent($name, $this->layoutObj, $properties)) {
-                throw new CmsException(Lang::get('cms::lang.component.not_found', ['name'=>$name]));
-            }
-
-            $componentObj->alias = $alias;
-            $this->vars[$alias] = $this->layout->components[$alias] = $componentObj;
+        if ($isSoftComponent = $this->isSoftComponent($name)) {
```

Comment:
@DanHarrin Could you take the variable assignment out of this condition and just check against the result? IMO, I find this sort of shortcut a bit of a code smell because it's easy to miss the single `=` when debugging.
```suggestion
        $isSoftComponent = $this->isSoftComponent($name);
        if ($isSoftComponent) {
```